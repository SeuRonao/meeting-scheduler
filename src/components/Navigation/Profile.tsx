import React from "react";
import { signOut } from "firebase/auth";
import { Stack, Button, Alert } from "react-bootstrap";
import { useAuthState } from "react-firebase-hooks/auth";
import { useTranslation } from "react-i18next";
import { useNavigate } from "react-router-dom";
import { auth } from "../../utils/firebase/firebaseInit";
import LoadingSpinner from "../common/LoadingSpinner";

export default function Profile() {
  const { t } = useTranslation();
  const [user, loading, error] = useAuthState(auth);
  const navigate = useNavigate();

  async function handleLogOut() {
    try {
      await signOut(auth);
      navigate("/");
    } catch (error) {
      console.log(error);
    }
  }

  if (error) {
    return <Alert variant="danger">{error.message}</Alert>;
  }

  if (user) {
    return (
      <Stack direction="horizontal" gap={2}>
        <span>{user.displayName || user.email}</span>
        <Button variant="outline-secondary" size="sm" onClick={handleLogOut}>
          {t("Profile.sign-out")}
        </Button>
      </Stack>
    );
  }

  if (loading) {
    return <LoadingSpinner />;
  }

  return (
    <Button variant="outline-success" href="/sign-in">
      {t("Profile.sign-in")}
    </Button>
  );
}
