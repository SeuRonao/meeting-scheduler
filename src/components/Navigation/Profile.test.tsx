import { MemoryRouter } from "react-router-dom";
import { create, act, ReactTestRenderer } from "react-test-renderer";
import Profile from "./Profile";

describe("Testing Profile", () => {
  test("renders correctly", async () => {
    let component: ReactTestRenderer;
    await act(async () => {
      component = create(
        <MemoryRouter>
          <Profile />
        </MemoryRouter>
      );
    });
    expect(component!.toJSON()).toMatchSnapshot();
  });
});
