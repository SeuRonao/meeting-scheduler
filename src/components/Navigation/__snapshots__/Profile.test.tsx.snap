// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Profile renders correctly 1`] = `
<a
  className="btn btn-outline-success"
  href="/sign-in"
  onClick={[Function]}
>
  sign-in
</a>
`;
